### ISSUE TITLE
NVDA and Firefox 36 - skiplink doesn't properly "land" on a button.

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#4907]
**Reported by digitaltoast on 2015-02-11 20:42**
Normal NVDA and Firefox play well together, but here's an oddity:

Visit http://kingston.talking-newspapers.co.uk/ either with a screen reader off, or with any other combination of NVDA and IE, Jaws and Firefox, Chromevox - pick and mix, your choice! 

Tab 3 times, activate the "skip straight to audio player" skiplink, and you'll land on a button. Tap space and that button will activate and play.

Now try it in NVDA and Firefox - it doesn't work. Additionally, once you DO manually navigate to the button, hitting space seems to activate the button twice. I've had users confirm the same with earlier version of Firefox, Window 7 etc.

Is there are error in the page markup (which passes the w3 validator) or is this an NVDA or Firefox problem?

Thanks!

**Comment by bhavyashah** [comment]
The provided URL seems to contain no button with NVDA 2017.2 and Firefox 54 stable. Pressing Enter on the Skip to audio player link results in the following:
Input: kb(laptop):enter
IO - speech.speak (11:44:42):
Speaking [LangChangeCommand ('en_GB'), u'Listen to the latest KTN Weekly from Kingston Upon Thames Talking Newspaper - Mozilla Firefox']
IO - speech.speak (11:44:42):
Speaking [LangChangeCommand ('en_GB'), u'unknown']
IO - speech.speak (11:44:42):
Speaking [LangChangeCommand ('en_GB'), u'Listen to the latest KTN Weekly from Kingston Upon Thames Talking Newspaper - Mozilla Firefox']
IO - speech.speak (11:44:42):
Speaking [LangChangeCommand ('en_GB'), u'Listen to the latest KTN Weekly from Kingston Upon Thames Talking Newspaper  document']
IO - speech.speak (11:44:42):
Speaking [LangChangeCommand ('en_GB'), u'visited  link', u'Skip straight to audio player or keep tabbing to choose different publications']

Since the provided test case may require subscription and isn't consequently serving its purpose, we have no means of contacting the original author of this ticket for an alternate test case, and it is possible that this may have been fixed between Firefox 36 to 54, I suggest closing. @ehollig 
@jcsteh On a slightly unrelated note, clicking the Skip to audio player results in odd and unexpected behaviour. Could you investigate that briefly and share your findings about what exactly is triggering that?

**Comment by ehollig** [comment]
This issue appears to work correctly in Internet Explorer and Chrome. It does not work as expected in Firefox. CC @jcsteh

**Comment by Adriani90** [comment]
What exactly is the issue here actually? I am testin in Firefox 64.0.2 with NVDA 2018.4.1. Pressing tab until skip to video, then pressing enter and then space. The video starts. Pressing space again, the video stops. The only issue I see here is that the button "play / pause" is not reported when activating the skiplink. You must press tab and shift + tab to get the button reported by NVDA.
However, in my opinion focus mode navigation on websites is anyway not working reliably in many cases. So browse mode navigation would be a better option here. Still, I wish the focus mode navigation would become more stable in general.

**Comment by Adriani90** [comment]
@MarcoZehe could you please help here? Honnestly I think I don'treally understand the issue. If this is not present in Firefox, we can close it because it seems we lost contact to the initial author.

**Comment by josephsl** [comment]
Hi, and FF36 is way old by today’s standards. Thanks.

**Comment by MarcoZehe** [comment]
That page has changed significantly since this report was filed. The player is the first skip link, and it navigates to the Media Player heading level 1 now. And that works just fine with NVDA and Firefox. This report can be closed.

**Comment by Adriani90** [comment]
Closing as works for me.
